bpf/cpu/cpu.bpf.c

```diff
@@ -1,47 +1,43 @@
 // +build ignore
```

Comment:
The changes in this file are huge. I did my best to make sense of all the things and catch issues, but it might not be a bulletproof review. 

So it's best to check out the result, if it works it works :) I trust you on this.

Reply:
Sorry about this! LMK if you spot anything that doesn't make sense or that's not clear. If it's not clear enough, it's a bug and that's on me :)